File: src/Stats.java
public class Stats {

	public static double max (int [] num) {
		if (num == null || num.length < 1) {
			return 0.0;
		}
		double max = num[0];
		for (int i = 0; i < num.length; i++) {
			if (num[i] > max)
				max = num [i];
		}
		return max;
	}

	public static double max (double [] num) {
		if (num == null || num.length < 1) {
			return 0.0;
		}
		double max = num[0];
		for (int i = 0; i < num.length; i++) {
			if (num[i] > max)
				max = num [i];
		}
		return max;
	}

	public static double min (int [] num) {
		if (num == null || num.length < 1) {
			return 0.0;
		}
		double min = num[0];
		for (int i = 0; i < num.length; i++) {
			if (num[i] < min)
				min = num [i];
		}
		return min;
	}

	public static double min (double [] num) {
		if (num == null || num.length < 1) {
			return 0.0;
		}
		double min = num[0];
		for (int i = 0; i < num.length; i++) {
			if (num[i] < min)
				min = num [i];
		}
		return min;
	}

	public static double avg (int [] num) {
		if (num == null || num.length < 1) {
			return 0.0;
		}
		double sum = 0.0;
		for (int i = 0; i < num.length; i++) {
			sum += num[i];
		}
		return sum/num.length;
	}

	public static double avg (double [] num) {
		if (num == null || num.length < 1) {
			return 0.0;
		}
		double sum = 0.0;
		for (int i = 0; i < num.length; i++) {
			sum += num[i];
		}
		return sum/num.length;
	}
}

File: src/stats.go
package stats

func average (data [] int) float32 {
	sum := data[0]
	for i:= 1; i < len (data); i++ {
		sum = sum + data [i]
	}
	var mean float32 = sum * 1.0 / len (data)
	return mean
}

func max (data [] int) int {
	if len (data) == 0 {
		return 0
	} else {
		max := data[0]
		for i:= 1; i < len (data); i++ {
			if data[i] > max {
				max = data[i]
			}
		}
		return max
	}
}

func min (data [] int) int {
	if len (data) == 0 {
		return 0
	} else {
		min := data[0]
		for i:= 1; i < len (data); i++ {
			if data[i] < min {
				min = data[i]
			}
		}
		return min
	}
}
